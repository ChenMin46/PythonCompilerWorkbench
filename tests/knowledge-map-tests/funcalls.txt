foo()
foo(1)
foo(1, 2, 3)
foo(x)
foo(y, z)
foo(x + y / z)
foo(bar(x))
spam(x=2, y = 3) # only keyword args
spam(0, 1, x=2, y=3) # regular and keyword args
foo(*x)
foo(**y)
foo(*x, **y) # star and kwargs
spam(   1,  x=2  ,  *y  ,    **z) # all four kinds of args!
