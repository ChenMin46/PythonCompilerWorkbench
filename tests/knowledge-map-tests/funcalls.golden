foo()
foo(1)
foo(1, 2, 3)
foo(x)
foo(y, z)
foo(x + y / z)
foo(bar(x))
spam(x=2, y = 3) # only keyword args
spam(0, 1, x=2, y=3) # regular and keyword args
foo(*x)
foo(**y)
foo(*x, **y) # star and kwargs
spam(   1,  x=2  ,  *y  ,    **z) # all four kinds of args!
===
Module
  body: [
    Expr
      value: Call >> 'foo()'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'foo(1)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
          Num >> '1'
            n: 1
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'foo(1, 2, 3)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
          Num >> '1'
            n: 1
          Num >> '2'
            n: 2
          Num >> '3'
            n: 3
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'foo(x)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
          Name >> 'x'
            id: 'x'
            ctx: Load
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'foo(y, z)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
          Name >> 'y'
            id: 'y'
            ctx: Load
          Name >> 'z'
            id: 'z'
            ctx: Load
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'foo(x + y / z)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
          BinOp >> 'x + y / z'
            left: Name >> 'x'
              id: 'x'
              ctx: Load
            op: Add
            right: BinOp >> 'y / z'
              left: Name >> 'y'
                id: 'y'
                ctx: Load
              op: Div
              right: Name >> 'z'
                id: 'z'
                ctx: Load
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'foo(bar(x))'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
          Call >> 'bar(x)'
            func: Name >> 'bar'
              id: 'bar'
              ctx: Load
            args: [
              Name >> 'x'
                id: 'x'
                ctx: Load
            ]
            keywords: [
            ]
            starargs: None
            kwargs: None
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'spam(x=2, y = 3)'
        func: Name >> 'spam'
          id: 'spam'
          ctx: Load
        args: [
        ]
        keywords: [
          keyword >> '2'
            arg: 'x'
            value: Num >> '2'
              n: 2
          keyword >> '3'
            arg: 'y'
            value: Num >> '3'
              n: 3
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'spam(0, 1, x=2, y=3)'
        func: Name >> 'spam'
          id: 'spam'
          ctx: Load
        args: [
          Num >> '0'
            n: 0
          Num >> '1'
            n: 1
        ]
        keywords: [
          keyword >> '2'
            arg: 'x'
            value: Num >> '2'
              n: 2
          keyword >> '3'
            arg: 'y'
            value: Num >> '3'
              n: 3
        ]
        starargs: None
        kwargs: None
    Expr
      value: Call >> 'foo(*x)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
        ]
        keywords: [
        ]
        starargs: Name >> 'x'
          id: 'x'
          ctx: Load
        kwargs: None
    Expr
      value: Call >> 'foo(**y)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
        ]
        keywords: [
        ]
        starargs: None
        kwargs: Name >> 'y'
          id: 'y'
          ctx: Load
    Expr
      value: Call >> 'foo(*x, **y)'
        func: Name >> 'foo'
          id: 'foo'
          ctx: Load
        args: [
        ]
        keywords: [
        ]
        starargs: Name >> 'x'
          id: 'x'
          ctx: Load
        kwargs: Name >> 'y'
          id: 'y'
          ctx: Load
    Expr
      value: Call >> 'spam(   1,  x=2  ,  *y  ,    **z)'
        func: Name >> 'spam'
          id: 'spam'
          ctx: Load
        args: [
          Num >> '1'
            n: 1
        ]
        keywords: [
          keyword >> '2'
            arg: 'x'
            value: Num >> '2'
              n: 2
        ]
        starargs: Name >> 'y'
          id: 'y'
          ctx: Load
        kwargs: Name >> 'z'
          id: 'z'
          ctx: Load
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "()"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "1"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "1"
                }, 
                {
                  "id": "tid_1", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "2"
                }, 
                {
                  "id": "tid_2", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "3"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "x"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "y"
                }, 
                {
                  "id": "tid_3", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "z"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(", 
                {
                  "contents": [
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Name", 
                      "value": "x"
                    }, 
                    " ", 
                    {
                      "id": "tid_4", 
                      "name": "token", 
                      "value": "+"
                    }, 
                    " ", 
                    {
                      "contents": [
                        {
                          "ctx": "Load", 
                          "id": "id_0", 
                          "name": "Name", 
                          "value": "y"
                        }, 
                        " ", 
                        {
                          "id": "tid_5", 
                          "name": "token", 
                          "value": "/"
                        }, 
                        " ", 
                        {
                          "ctx": "Load", 
                          "id": "id_0", 
                          "name": "Name", 
                          "value": "z"
                        }
                      ], 
                      "id": "id_0", 
                      "name": "BinOp", 
                      "op": "Div"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "BinOp", 
                  "op": "Add"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(", 
                {
                  "contents": [
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Name", 
                      "value": "bar"
                    }, 
                    "(", 
                    {
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Name", 
                      "value": "x"
                    }, 
                    ")"
                  ], 
                  "id": "id_0", 
                  "name": "Call"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "spam"
                }, 
                "(x=", 
                {
                  "arg": "x", 
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "2"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "keyword"
                }, 
                ", y = ", 
                {
                  "arg": "y", 
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "3"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "keyword"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        " ", 
        {
          "id": "tid_6", 
          "name": "comment", 
          "value": "# only keyword args"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "spam"
                }, 
                "(", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "0"
                }, 
                {
                  "id": "tid_7", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "1"
                }, 
                ", x=", 
                {
                  "arg": "x", 
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "2"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "keyword"
                }, 
                ", y=", 
                {
                  "arg": "y", 
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "3"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "keyword"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        " ", 
        {
          "id": "tid_8", 
          "name": "comment", 
          "value": "# regular and keyword args"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(*", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "x"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(**", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "y"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "(*", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "x"
                }, 
                ", **", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "y"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        " ", 
        {
          "id": "tid_9", 
          "name": "comment", 
          "value": "# star and kwargs"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "spam"
                }, 
                "(   ", 
                {
                  "id": "id_0", 
                  "name": "Num", 
                  "value": "1"
                }, 
                ",  x=", 
                {
                  "arg": "x", 
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "2"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "keyword"
                }, 
                "  ,  *", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "y"
                }, 
                "  ,    **", 
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "z"
                }, 
                ")"
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    " ", 
    {
      "id": "tid_10", 
      "name": "comment", 
      "value": "# all four kinds of args!"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
